code/obj/storage/vis_storage_controller.dm

```diff
@@ -0,0 +1,22 @@
+/datum/vis_storage_controller
+	var/obj/parent = null
+	///The items this storage controller contains
+	var/list/obj/vis_items = list()
+
+	proc/initialize(new_parent)
+		parent = new_parent
```

Comment:
This could be achieved by passing the parent atom to `new()` instead of implementing a separate proc for it.

```
/datum/vis_storage_controller/New(obj/parent)
    . = ..()
    src.parent = parent
```